### PR TITLE
Remove flawed TTL API

### PR DESCRIPTION
**Pull request by Elesbaan70**
This is my solution to the problem of false and unmaintainable traffic light API, as discussed here: https://github.com/CitiesSkylinesMods/TMPE/issues/967#issuecomment-1073163111

If this or some similar solution is not used, the ability to enhance traffic lights will always be hampered by the need to carry along a fundamentally defective API. It is probably impossible to maintain API compatibility with enhancements of any size, and so the time to nuke the API is now, when we believe that there are no other mods using it.

### CODE REVIEW

**Comment by Elesbaan70** [comment]
@krzychu124 I created a new clone of my repository so I would be starting clean from what's in GitHub, and it built fine. Are you able to build this?

File: TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs
Comment (kianzarrin) [inline TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:176]: [Note] I think TMPE policy is that using this is recommended. I don't care either way

File: TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs
Comment (kianzarrin) [inline TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:1693]: [Note]
I think your auto code cleanup has been setup to do some extra stuff. 
I also think its `.editorConfig` fault for not explicitly handling this situations and leaving it to the person's IDE settings.

Comment (kianzarrin) [reply TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:1693]: it just means there is more code to review.

Comment (kianzarrin) [reply TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:1693]: and increases chance of conflict.

**Review by kianzarrin — APPROVED** [review]
LGTM. just renames.

@Elesbaan70 can you please test persistency?

File: TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs
Comment (Elesbaan70) [reply TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:1693]: Yes, I did not tell Visual Studio to do any of that, nor did I explicitly configure it to happen automatically. It's extremely frustrating. But I can go back and revert these if y'all believe it's worthwhile.

File: TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs
Comment (Elesbaan70) [reply TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:176]: This was Visual Studio doing things I didn't ask it to. :-(

File: TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs
Comment (kianzarrin) [reply TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:1693]: No I don't think its worth while.
But maybe you can play with .editorconfig not to do this. But I think that's outside of scope of this PR

**Comment by Elesbaan70** [comment]
> I've done basic testing already, but I've decided to go back and do more. Did you have a specific areas of concern?

File: TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs
Comment (kianzarrin) [reply TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:1693]: image
for some reason you have to use text editor to modify values.

**Review by aubergine10 — COMMENTED** [review]
LGTM 👍

**Comment by Elesbaan70** [comment]
> @Elesbaan70 can you please test persistency?

Persistency is forward- and backward-compatible. Reverting to stable does not work because of #1492 and the versioning issue that will be addressed by #1502. However, reverting to the master branch works fine.

**Comment by Elesbaan70** [comment]
On another note, I'm not sure what I should think about the possibility that my first contribution might be nuking an entire API. LOL

**Comment by aubergine10** [comment]
@Elesbaan70 pls update to latest master branch 
image

**Comment by aubergine10** [comment]
Ugh, @Elesbaan70 I messed up your PR with recent merged PR - want me to send some commits to fix the merge conflicts?

File: TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs
Comment (Elesbaan70) [reply TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:1822]: GitHub won't let me view the whole diff on mobile, so I'll have to confirm later, but this looks like a glitch in the diff. Notice how the whole line doesn't match up, not just the part it marked as changed.

**Comment by Elesbaan70** [comment]
> want me to send some commits to fix the merge conflicts?

If you already have it, sure. I suspect I can just do another merge and clean things up pretty easily, in any case.

File: TLM/TMPE.API/TMPE.API.csproj
Comment (aubergine10) [inline TLM/TMPE.API/TMPE.API.csproj:183]: ?

**Comment by aubergine10** [comment]
@Elesbaan70 I've done most of the conflicts from the `IsStartNode` PR so hopefully that will reduce amount of stuff you need to wade through.

File: TLM/TMPE.API/TMPE.API.csproj
Comment (krzychu124) [reply TLM/TMPE.API/TMPE.API.csproj:183]:
```suggestion
```

File: TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs
Comment (Elesbaan70) [reply TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:1822]: This has got to be something in TMPE's project settings, because I've never had Visual Studio go through and refactor things without me asking it to.

Comment (krzychu124) [reply TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:1822]: Yeah, looks good. The problem was in weird `switch..case` whitespace
```cs
switch(xyz) {
....case 1: {
   // normal closing bracket
....}
}
```
```cs
switch(xyz) {
....case 1: {
   // closing bracket in this PR (surprising - seem to be only applied to switch..case)
........}
}
```

File: TLM/TMPE.API/TMPE.API.csproj
Comment (Elesbaan70) [reply TLM/TMPE.API/TMPE.API.csproj:183]: We could remove it. When it comes to generated files, I tend to just let Visual Studio do its thing unless I have a specific concern.

**Comment by Elesbaan70** [comment]
> ok, just fix conflicts :)

Yep, I'll be on that this evening.

File: TLM/TMPE.API/TMPE.API.csproj
Comment (aubergine10) [reply TLM/TMPE.API/TMPE.API.csproj:183]: remove imo

Comment (Elesbaan70) [reply TLM/TMPE.API/TMPE.API.csproj]: Done

**Comment by Elesbaan70** [comment]
@aubergine10 Thanks. I did end up reverting it, though. When I used a decent merge tool, it only found one conflict that it couldn't resolve automatically.

**Review by aubergine10 — APPROVED** [review]
LGTM 👍

**Comment by Elesbaan70** [comment]
I highly recommend P4Merge for when merges get messy. https://www.perforce.com/downloads/visual-merge-tool

Watch out with the installer, though. It will try to install their whole version control client, so you have to uncheck everything except the P4Merge option.

**Comment by aubergine10** [comment]
@Elesbaan70 pls merge this in to `master` (assuming you can now it's got 2+ approvals)

**Comment by Elesbaan70** [comment]
I don't have write access

**Pull request merged**
